src/cswinrt/code_writers.h

```diff
@@ -453,6 +454,82 @@ namespace cswinrt
             bind<write_parameter_name>(param));
     }
 
+    void write_projection_arg(writer& w, method_signature::param_t const& param)
+    {
+        w.write("%",
+            bind<write_parameter_name>(param));
+    }
+
+    void write_event_source_type_name(writer& w, type_semantics const& eventTypeSemantics)
+    {
+        auto eventTypeCode = w.write_temp("%", bind<write_type_name>(eventTypeSemantics, typedef_name_type::Projected, false));
+        std::string eventTypeName = "_EventSource_" + eventTypeCode;
+        std::regex re("(\\ |\\:|\\<|\\>|\\,|\\.)");
+        w.write("%", std::regex_replace(eventTypeName, re, "_"));
+    }
+
+    void write_event_invoke_params(writer& w, TypeDef const& eventType)
+    {
+        for (auto&& method : eventType.MethodList())
```

Comment:
consider factoring out lookup of the "Invoke" method and binding all your write_event_invoke_* methods to that instead of eventType - simpler and faster code.